docs/user/reference/general/initializing.md

```diff
@@ -1,39 +1,80 @@
 # Initializing
 
-The following steps are required for applications using the Glean SDK, but not libraries.
+Glean needs to be initialized in order to be able to send pings,
+record metrics and perform maintenance tasks. Thus is is advised that Glean be initialized as
+soon as possible in an application's lifetime and importantly, before any other libraries in the
+application start using Glean.
 
 {{#include ../../../shared/blockquote-info.html}}
 
-##### Note
+## Libraries are not required to initialize Glean
 
-> The `initialize` function _must_ be called, even if telemetry upload is disabled.
-> Glean needs to perform maintenance tasks even when telemetry is disabled, and because Glean
-> does this as part of its initialization, it is _required_ to always call the `initialize`
-> function. Otherwise, Glean won't be able to clean up collected data, disable queuing of pre-init
-> tasks, or perform other required operations.
->
-> This does not apply to special builds where telemetry is disabled at build time. In that case, it is acceptable to not call `initialize` at all.
+> Libraries rely on the same Glean singleton as the application in which they are
+> embedded. Hence, they are not expected to initialize Glean
+> as the application should already do that.
+
+## Behavior when uninitialized
+
+Any API called before Glean is initialized is queued and applied at initialization.
+To avoid unbounded memory growth the queue is bounded (currently to a maximum of 100 tasks),
+and further calls are dropped.
+
+The number of calls dropped, if any,
+is recorded in the `glean.error.preinit_tasks_overflow` metric.
+
+## Behavior once initialized
+
+### When upload is enabled
+
+Once initialized, if upload is enabled,
+Glean will automatically start applying enqueued tasks, such as
+collection of [baseline metrics](../../user/pings/metrics.md).
+
+### When upload is disabled
+
+If upload is disabled, any persisted metrics, events and pings (other than `first_run_date` and `first_run_hour`) are cleared, any pending deletion-request is sent and subsequent calls to record metrics and submit pings will be no-ops. Because Glean does that as part of its initialization, users are _required_ to always initialize Glean.
```

Comment:
Add some newlines for easier editing and a bit of restructuring.

```suggestion
If upload is disabled, any persisted metrics, events and pings (other than `first_run_date` and `first_run_hour`) are cleared.
Pending `deletion-request` pings are sent.
Subsequent calls to record metrics and submit pings will be no-ops.
Because Glean does that as part of its initialization, users are _required_ to always initialize Glean.
```